### PR TITLE
API Implement Lookup class, convert major branches to minors, "smart" namespace lookups

### LINKED ISSUE #33: /search/lookup URLs broken
From @spiritlevel

> The parsing of [api:Director] style short codes seems to be generating the correct query:
> api.silverstripe.org/search/lookup/?q=Director&version=3&module=framework
> The problem is that the api site no longer processes such queries. These queries used to work :(

### PR DESCRIPTION
**Pull request by robbieaverill**
Replaces #37 and would require #36 (although easily changed).

Fixes #33, resolves https://github.com/silverstripe/silverstripe-docsviewer/issues/132, fixes https://github.com/silverstripe/silverstripe-docsviewer/issues/129 and resolves https://github.com/silverstripe/api.silverstripe.org/issues/32

### CODE REVIEW

**Comment by robbieaverill** [comment]
Also this PR adds Travis support - will need someone in @silverstripe/open-sourcerers to enable it

**Comment by tractorcow** [comment]
ok setup at https://travis-ci.org/silverstripe/api.silverstripe.org

File: README.md [hunk 1]
```diff
@@ -56,11 +59,13 @@ Parameters:
...
+ * `version`: (optional) Version of the targeted module. Should map to a folder name. Defaults to trunk. Will switch current unstable major version (e.g. 4) to "master".
  * `module`: (optional) Module name. Should map to a folder name. Defaults to framework.
 
 Examples:
 
  * `/search/lookup.php?q=DataObject`: Shows `DataObject` docs in `trunk` version of framework
  * `/search/lookup.php?q=DataObject::get()&version=3.0`: Shows `DataObject::get()` docs in `3.0` version of framework
- * `/search/lookup.php?q=DPSPayment&module=payemtn`: Shows `DPSPayment` class docs in the `ecommerce` module
\ No newline at end of file
+ * `/search/lookup.php?q=DataObject::get()&version=3.0`: Shows `DataObject::get()` docs in `3.6` version of framework (or whatever is the latest stable minor version)
```
Comment (robbieaverill) [inline README.md]: For now we're still on minor branches for the API docs, so 3.6 is correct.

**Comment by robbieaverill** [comment]
@SpiritLevel good spot on those typos. Fancy a PR to fix them?

**Comment by robbieaverill** [comment]
Fixed in #39

**Pull request merged**